src/DeFuncto.Tests/Core/Types/Du5/Match.cs

```diff
@@ -1,59 +1,61 @@
 ﻿using DeFuncto.Extensions;
+using FsCheck;
+using FsCheck.Xunit;
 using Xunit;
 using static DeFuncto.Prelude;
 
 namespace DeFuncto.Tests.Core.Types.Du5
 {
     public class Match
     {
-        [Fact(DisplayName = "Matches the First option")]
-        public void OnFirst() =>
-            First<string, int, int, int, int>("ban").Match(
-                val => $"{val}ana",
-                val => $"{val}ana",
-                val => $"{val}ana",
-                val => $"{val}ana",
-                val => $"{val}ana"
-            ).Run(result => Assert.Equal("banana", result));
+        [Property(DisplayName = "Matches the First option")]
```

Comment:
For all of these cases, add a third input for the ones that are not being matched.